bundle/keys.go

```diff
@@ -107,17 +108,24 @@ func (s *SigningConfig) WithPlugin(plugin string) *SigningConfig {
 // GetPrivateKey returns the private key or secret from the signing config
 func (s *SigningConfig) GetPrivateKey() (interface{}, error) {
 	var priv string
-	if _, err := os.Stat(s.Key); err == nil {
-		bs, err := ioutil.ReadFile(s.Key)
-		if err != nil {
+
+	block, _ := pem.Decode([]byte(s.Key))
+	if block == nil {
+		if _, err := os.Stat(s.Key); err == nil {
+			bs, err := ioutil.ReadFile(s.Key)
+			if err != nil {
+				return nil, err
+			}
+			priv = string(bs)
+		} else if os.IsNotExist(err) {
+			priv = s.Key
+		} else {
```

Comment:
[nit] We could invert branches a little and cut down the indentation, I believe:
```go
func (s *SigningConfig) GetPrivateKey() (interface{}, error) {
	block, _ := pem.Decode([]byte(s.Key))
	if block != nil { // it's a valid key
		return sign.GetSigningKey(priv, jwa.SignatureAlgorithm(s.Algorithm))
	}
	
	if _, err := os.Stat(s.Key); err != nil { // not a key, not a file, error out
		return nil, err
	}
	
	bs, err := ioutil.ReadFile(s.Key)
	if err != nil {
		return nil, err
	}
	return sign.GetSigningKey(string(bs), jwa.SignatureAlgorithm(s.Algorithm))
}
```

What do you think? IMHO it's slightly easier to understand what's going on here, but it's not a big deal either way.

Also, we're throwing away work -- we'll parse a valid key PEM string twice. We control `sign.GetSigningKey`, so if this was a concern, we could change the interface there 🤔

Reply:
Updated the code a little as per your suggestion.

Reply:
Hrmm I think this bit is still superfluous, isn't it? 
https://github.com/open-policy-agent/opa/pull/3772/files#diff-294071c71eef316baabc3d7701313ee25faf7da311f56565b2f3eea59e09cf84L116

If the error is "not found", and it's supposed to be a key string, it's already been dealt with at that point.

Reply:
> Hrmm I think this bit is still superfluous, isn't it?
> https://github.com/open-policy-agent/opa/pull/3772/files#diff-294071c71eef316baabc3d7701313ee25faf7da311f56565b2f3eea59e09cf84L116
> 
> If the error is "not found", and it's supposed to be a key string, it's already been dealt with at that point.

It could also be a secret (HMAC) for instance.

Reply:
Ah okay I wasn't aware. Thanks.

Reply:
We can't have an hmac secret of "/dev/zero" or something like that on unixy platforms, then, hm? I think I'd prefer having two config keys, one for verbatim X and one for file name X things. 🤔 Just to avoid ambiguity.

Reply:
Do you mind filing an issue to track this ?